docs/index.rst

```diff
@@ -0,0 +1,17 @@
+################################
+Qiskit Algorithms documentation
+################################
+
+
```

Comment:
I guess this really ought to have something in it as otherwise one is looking more or less a blank page when things start out. An overview for example like we have in the apps.

We should also have a Getting Started too.

(Once CI is running ok with what is already here of course!)